streaming/src/main/scala/org/apache/spark/streaming/receiver/ReceiverSupervisorImpl.scala

```diff
@@ -73,14 +73,16 @@ private[streaming] class ReceiverSupervisorImpl(
 
   /** Timeout for Akka actor messages */
   private val askTimeout = AkkaUtils.askTimeout(env.conf)
+  private var numRegisterAttemps = -1
 
   /** Akka actor for receiving messages from the ReceiverTracker in the driver */
   private val actor = env.actorSystem.actorOf(
     Props(new Actor {
       override def preStart() {
-        logInfo("Registered receiver " + streamId)
-        val msg = RegisterReceiver(
-          streamId, receiver.getClass.getSimpleName, Utils.localHostName(), self)
+        numRegisterAttemps += 1
+        logInfo("Register receiver " + streamId + " before it starts ")
+        val msg = RegisterReceiver(streamId, receiver.getClass.getSimpleName,
```

Comment:
@jerryshao yeah, I am confused too.  IMHO, it is better to distinguish "register state" and "start state".